core/src/services/obs/core.rs

```diff
@@ -254,4 +317,16 @@ impl ObsCore {
 
         self.send(req).await
     }
+
+    pub async fn sign_query<T>(&self, req: &mut Request<T>, duration: Duration) -> Result<()> {
```

Comment:
Can you please put this method together with `sign`? Thank you.

https://github.com/apache/incubator-opendal/blob/5552c9741a754067d54d1f2c065fc79bc6871add/core/src/services/obs/core.rs#L70